src/datasources/class-array-datasource.ts

```diff
@@ -0,0 +1,17 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { BaseOpenMRSDataSource } from './data-source';
+
+export class ClassArrayDataSource extends BaseOpenMRSDataSource {
+  constructor() {
+    super('/ws/rest/v1/concept?v=custom:(uuid,display,conceptClass:(uuid,display))');
+  }
+
+  fetchData(searchTerm: string, config?: Record<string, any>): Promise<any[]> {
+    return openmrsFetch(searchTerm ? `${this.url}&q=${searchTerm}` : this.url).then(({ data }) => {
+      if(config?.class){
+        return data.results.filter(concept => concept.conceptClass && config.class.includes(concept.conceptClass.uuid))
+      }
+      return data.results;
+    });
+  }
+}
```

Comment:
@kajambiya how is this data source different from the concept data source?

Reply:
So I wanted concept datasource to be used for fetching either all concepts or concepts by class if one specified the class. For example, for drug, we pass the drug class in the url when calling the api and fetch all concepts of class drug.

Class array datasource (naming could be improved) on the other hand was to fetch concepts from more than one class. Since we can't pass an array of classes to the url, the logic is to first fetch all concepts and then filter out those that belong to the specified classes. That's why I separated the 2 so that each can handle the certain type of use cases.

Reply:
Had preferred to separate concerns so that there are as minimal conditions as possible but let me make the update and you take a look

Reply:
I agree with @ebambo on this. We can simply make the concept ds more configurable. It can accept an array of concept classes. If the length of the concept classes is > 1, we apply a filter to the search results. If it's simply one we apply the concept class as a query parameter.

Reply:
@eudson @samuelmale 
I've refactored the code accordingly. Please review and revert.